components/esp_modem/examples/modem_psm/main/modem_psm.c

```diff
@@ -0,0 +1,222 @@
+
+    /* By AT command */
+    CHECK_ERR(esp_modem_at(dce, "AT+CPOWD=1", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+
+    /* By setting the power key */
+    //rtc_gpio_hold_dis(GPIO_OUTPUT_PWRKEY);
+    //gpio_set_level(GPIO_OUTPUT_PWRKEY, 1);
+    //vTaskDelay(pdMS_TO_TICKS(2000));
+    //gpio_set_level(GPIO_OUTPUT_PWRKEY, 0);
+    //rtc_gpio_hold_en(GPIO_OUTPUT_PWRKEY);
+    //read_data(data);
+}
+
+void run_at_cmd(esp_modem_dce_t *dce)
+{
+    char data[BUF_SIZE];
+    CHECK_ERR(esp_modem_at(dce, "AT+CSQ", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+    vTaskDelay(pdMS_TO_TICKS(1000));
+
+    CHECK_ERR(esp_modem_at(dce, "AT+CPSMS?", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+    vTaskDelay(pdMS_TO_TICKS(1000));
+
+    CHECK_ERR(esp_modem_at(dce, "AT+CPSI?", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+    CHECK_ERR(esp_modem_at(dce, "AT+CNACT=0,1", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+    CHECK_ERR(esp_modem_at(dce, "AT+SNPDPID=0", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+    CHECK_ERR(esp_modem_at(dce, "AT+SNPING4=\"8.8.8.8\",3,16,1000", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+    CHECK_ERR(esp_modem_at(dce, "AT+CNACT=0,0", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+}
+
+
+void app_main(void)
+{
+    char *cmd;
+    char data[BUF_SIZE];
+    esp_modem_dce_t *dce = NULL;
+    esp_netif_t *esp_netif = NULL;
+    esp_sleep_wakeup_cause_t wakeup_cause = esp_sleep_get_wakeup_cause();
+    ESP_LOGI(TAG, "Deep sleep Wake Up Cause: %d\n", wakeup_cause);
+
+    ESP_ERROR_CHECK(esp_netif_init());
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
+
+    switch (wakeup_cause) {
+    case ESP_SLEEP_WAKEUP_UNDEFINED:
+        uart_init(&dce, &esp_netif);
+
+        /* Power on the modem */
+        config_pwrkey_gpio();
+        power_on_modem(dce);
+
+        for (int i = 0; i < 3; i++) {
+            memset(data, 0, BUF_SIZE);
+            CHECK_ERR(esp_modem_at(dce, "AT", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+            vTaskDelay(pdMS_TO_TICKS(1000));
+        }
+
+        /* Configure Power Saving Mode in the modem */
+        asprintf(&cmd, "AT+CPSMS=1,,,\"%s\",\"%s\"\r\n", CONFIG_EXAMPLE_MODEM_T3412_PERIODIC_TAU,
+                 CONFIG_EXAMPLE_MODEM_T3324_ACTIVE_TIME);
+        CHECK_ERR(esp_modem_at(dce, cmd, data, 500), ESP_LOGI(TAG, "OK. %s", data));
+        free(cmd);
+        vTaskDelay(pdMS_TO_TICKS(1000));
+
+        /* Perform Communication tasks */
+        run_at_cmd(dce);
+        vTaskDelay(pdMS_TO_TICKS(1000));
+        break;
+
+    case ESP_SLEEP_WAKEUP_EXT0:
+        uart_init(&dce, &esp_netif);
+
+        for (int i = 0; i < 3; i++) {
+            memset(data, 0, BUF_SIZE);
+            CHECK_ERR(esp_modem_at(dce, "AT", data, 500), ESP_LOGI(TAG, "OK. %s", data));
+            vTaskDelay(pdMS_TO_TICKS(1000));
+        }
+
+        /* Perform Communication tasks */
+        run_at_cmd(dce);
+        vTaskDelay(pdMS_TO_TICKS(1000));
+        break;
+
+    case ESP_SLEEP_WAKEUP_TIMER:
```

Comment:
Question: Is this still valid? I thought we're to wake up only on ext interrupt.